src/core/lib/gprpp/debug_location.h

```diff
@@ -38,7 +38,7 @@ class DebugLocation {
   const char* file_;
   const int line_;
 };
-#define DEBUG_LOCATION ::grpc_core::DebugLocation(__FILE__, __LINE__)
+#define DEBUG_LOCATION grpc_core::DebugLocation(__FILE__, __LINE__)
```

Comment:
Macros should always use the fully qualified name: we don't control where they'll be expanded.

For instance... the following customer code would now fail:
```
#include "src/core/lib/gprpp/debug_location.h"
namespace sprockets_co {
namespace grpc_core { class DebugLocation {}; };
void foo() { 
  /* gets sprockets_co::grpc_core::DebugLocation(), not ::grpc_core::DebugLocation() */ 
  CallSomething(DEBUG_LOCATION); 
}

Reply:
This PR automerged after an LGTM, I've addressed this in https://github.com/grpc/grpc/pull/28917. In this PR, I had added logic to ignore `#define` statements in the `check_redundant_namespace_qualifiers.py` script for exactly that reason, but neglected to do so on the `check_namespace_qualification.py` script as well.